tests/test_freezing/test_freezing_compare.py

```diff
@@ -0,0 +1,187 @@
+"""Tests for freezing_compare.py."""
+
+import pandas as pd
+from pandas._testing import assert_frame_equal
+
+from src.freezing.freezing_compare import get_amendments, get_additions
+
+
+class TestGetAmendments:
+    """Tests for get_amendments()"""
+    # Create test frozen df
+    def create_test_frozen_df(self) -> pd.DataFrame:
+        """Create a test frozen df."""
+        input_cols = ["reference", "period", "instance", "201", "202", "200", "601"]
+        data = [
+            ["A", 202412, 2.0, 1.0, 2.0, "A", None],
+            ["B", 202412, None, None, 1.0, "B", "C"],
+            ["C", 202412, 0.0, 1.0, 2.0, "A", "B"],
+            ["D", 202412, 1.0, 2.0, 3.0, "C", "D"],
+            ["E", 202412, None, 4.0, 5.0, "E", "F"],
+        ]
+        input_frozen_df = pd.DataFrame(data=data, columns=input_cols)
+
+        numeric_cols = [
+            "203", "204", "205", "206", "207", "209", "210",
+            "211", "212", "214", "216", "218", "219", "220", "221", "222",
+            "223", "225", "226", "227", "228", "229", "237", "242", "243",
+            "244", "245", "246", "247", "248", "249", "250", "405", "406",
+            "407", "408", "409", "410", "411", "412", "501", "502", "503",
+            "504", "505", "506", "507", "508", "602", "701", "702", "703",
+            "704", "705", "706", "707", "709", "711",
+        ]
```

Comment:
for this, you can do something as simple as this (inside or outside the class, up to you)

```python
@pytest.fixture(scope="function")
def numeric_cols(self[optional based on class or not]) -> list:
    """Numerical columns for test dataframes."""
    numeric_cols = []
    return numeric_cols
```

Reply:
to add the cols, you can also make a function in the class, something like _add_numeric_cols() which returns a passed dataframe with the new columns.

Reply:
Thank you!